From 41376955a5a92b83702f5fd471f22ce9bde3be2c Mon Sep 17 00:00:00 2001
From: Kuba Schenk <abukcz@gmail.com>
Date: Thu, 1 Dec 2016 21:48:26 +0100
Subject: [PATCH 18/32] OMS7-N: ApplicationsState: add filter for Substratum
 icon overlays [1/2]

This commit allows the framework to handle the filtering of the icon overlays found for OMS.

Base this off the work from @KreAch3R

Change-Id: I594c993977733e67f566ac65df50ad2e1bbdbdd3
---
 .../android/settingslib/applications/ApplicationsState.java   | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java b/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java
index 70a92d6..3ae984f 100644
--- a/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java
+++ b/packages/SettingsLib/src/com/android/settingslib/applications/ApplicationsState.java
@@ -1311,6 +1311,17 @@ public class ApplicationsState {
         }
     };
 
+    public static final AppFilter FILTER_SUBSTRATUM_ICONS = new AppFilter() {
+        public void init() {
+        }
+
+        @Override
+        public boolean filterApp(AppEntry entry) {
+            return !((entry.info.metaData != null) &&
+                    (entry.info.metaData.getString("Substratum_IconPack") != null));
+        }
+    };
+
     public static final AppFilter FILTER_WORK = new AppFilter() {
         private int mCurrentUser;
 
-- 
2.9.3

